core/geotime/src/main/java/mil/nga/giat/geowave/core/geotime/store/filter/SpatialQueryFilter.java

```diff
@@ -196,8 +204,73 @@ public void fromBinary(
 					fieldId));
 		}
 		buf.get(theRest);
-		queryGeometry = geometryInterner.intern(GeometryUtils.geometryFromBinary(geometryBinary));
+		// cannot intern the prepared geometry. It uses a Object.hashCode().
+		preparedGeometryImage = geometryImageInterner.intern(new GeometryImage(
+				geometryBinary));
+		preparedGeometryImage.init();
 
 		super.fromBinary(theRest);
 	}
+
+	/**
+	 * Used for interning Geometry. Interning a geometry provides no additional
+	 * benefit since the geometry is already constructed in the heap at the
+	 * interning begins. The byte image of geometry provides a more efficient
+	 * component to hash and associate with a single image of the geometry.
+	 * Prepared geometries cannot be interned since they do not extend
+	 * Object.hashCode().
+	 * 
+	 * Using this approach increased performance of a large query unit test by 40% and
+	 * eliminated out of heap memory exceptions during maven builds.
```

Comment:
IMO you need to clarify this comment more - I don't believe this iterator stack was benchmarked;  rather the comparison was using a preparedgeometry vs. a non prepared geometry.   Which was probably not where you were running out of heap - did that happen because you couldn't intern a prepared geometry?   I think that's two different cases though.